lib/rubyoctopus/octopusconnection.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+require "faraday"
+require "faraday_middleware"
+
+module RubyOctopus
+  # Base class to use to perform actions against an Octopus instance.
+  class OctopusConnection
+    # Users might want to check which domain an instance is targeting.
+    attr_reader :url
+
+    # No need to expose the connection object.
+    @conn = nil
```

Comment:
Is this i-`nil`-tialization necessary? My preference would be to leave it out and just declare it in `initialize`.

Reply:
Oh, I would assume it is not necessary; I started it as a reader but figured that wouldn't be needed and just changed it slightly.